sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -1805,6 +1805,8 @@ HAT: '^';
 STRING
     : '\'' ( ~('\''|'\\') | ('\\' .) )* '\''
     | '"' ( ~('"'|'\\') | ('\\' .) )* '"'
+    | 'R\'' (~'\'')* '\''
+    | 'R"'(~'"')* '"'
```

Comment:
shall we define a new token for it? then the parser can choose to not call `unescapeSQLString` for raw string token.

Reply:
If we have a new token aside from `STRING`, we need to care all the places in `SqlBase.g4` and `ASTBuilder.scala` where `STRING` is referred. Is it acceptable?